tests/backends/base/test_base.py

```diff
@@ -60,6 +61,28 @@ def test_check_database_version_supported_with_none_as_database_version(self):
         with patch.object(connection.features, "minimum_database_version", None):
             connection.check_database_version_supported()
 
+    def test_release_memory_without_garbage_collection(self):
+        # Clear the garbage list to avoid interfering with the test.
+        gc.collect()
+        # Turn off automatic garbage collection.
+        gc.disable()
+        # Set the garbage collection debugging flag to store all unreachable
+        # objects in `gc.garbage`.
+        gc.set_debug(gc.DEBUG_SAVEALL)
+
+        # Create and close a connection.
+        connection.copy().close()
+
+        # Enforce garbage collection.
+        gc.collect()
+
+        # Ensure that the garbage list is empty.
+        self.assertEqual(gc.garbage, [])
+
+        # Clean up the garbage collection settings.
+        gc.set_debug(0)
+        gc.enable()
```

Comment:
Issue: If any code before this line fails, the remainder of the process has no GC, which could cause huge problems. Probably best wrapping the GC changes in a `try: finally:` to ensure they're called correctly.